app/uk/gov/hmrc/excisemovementcontrolsystemapi/connectors/NrsConnectorNew.scala

```diff
@@ -67,10 +69,23 @@ class NrsConnectorNew @Inject() (
         responseStatusAsFailure()
       )
       .flatMap { response =>
-        if (!is5xx(response.status)) {
-          Future.successful(Done)
-        } else {
-          Future.failed(UnexpectedResponseException(response.status, response.body))
+
+        response.status match {
+          case status if is5xx(status) =>
+            logger.warn(
+              s"Non repudiation submission failed with status ${response.status}"
+            )
+            Future.failed(UnexpectedResponseException(response.status, response.body))
+          case ACCEPTED =>
```

Comment:
I'd possibly make this the first case (then it's good path then all bad paths)

Reply:
fair point!